Split up lint jobs

jasonaowen opened the pull request:
Run each linter in a separate GitHub Actions job, so that what exactly is wrong is more obvious at a glance, and to more easily add additional linters (like #321).

We will need to reconfigure our merge requirements, which are currently expecting the single job `lint`.

jasonaowen commented:
This is split off from #321.

slifty reviewed (approved):
Nice improvement thanks!

jasonaowen commented:
Thanks!

I disabled the lint job requirement in the `main` branch protection rules. Once all the open PRs have either been merged or rebased, we can enable the requirement for the new, split jobs.

jasonaowen commented:
Rebased onto latest `main`; I will merge when the checks are done.

slifty commented:
@jasonaowen I was starting to approve some dependabots but I'm going to hold off on that since it will put you in a "constantly rebase" state!

jasonaowen commented:
@slifty ah, thank you, that would! Rebased again.

jasonaowen commented:
> I disabled the lint job requirement in the `main` branch protection rules. Once all the open PRs have either been merged or rebased, we can enable the requirement for the new, split jobs.

Since we require pull request branches to be up to date - that is, we require rebasing onto latest `main` to make the merge button turn green - I updated the branch protection to make all the new lint jobs required.
